extensions/apps/settings/src/components/pages/notifications-preferences-option.tsx

```diff
@@ -28,6 +95,65 @@ const NotificationsPreferencesOption: React.FC = () => {
     });
   };
 
+  const handleUnlockPreferences = async () => {
+    const enabled = await enableNotifications();
+    if (!enabled) {
+      _uiEvents.current.next({
+        event: NotificationEvents.ShowNotification,
+        data: {
+          type: NotificationTypes.Error,
+          title: t('Couldn’t unlock preferences'),
+          description: enabled ? undefined : t('Signature verification failed. Please try again.'),
+        },
+      });
+    }
+  };
+
+  const handleToggleAll = (val: boolean) => {
+    setPreferences(preferencesObjectFactory(val));
+    setEnableAllChecked(val);
+  };
+
+  const handleSetAntenna = (value: boolean) => {
+    setPreferences(prevState =>
+      prevState.map((item, idx) =>
+        idx === ANTENNA_ARR_INDEX ? { ...item, enabled: value } : item,
+      ),
+    );
+  };
+
+  const handleSetProfile = (value: boolean) => {
+    setPreferences(prevState =>
+      prevState.map((item, idx) =>
+        idx === PROFILE_ARR_INDEX ? { ...item, enabled: value } : item,
+      ),
+    );
+  };
```

Comment:
Both can be replaced by one handler something like:
 ```
const handleTogglePreference = (value: boolean, index: number) => {
   setPreferences(prevState =>
      prevState.map((item, idx) =>
        idx === index ? { ...item, enabled: value } : item,
      ),
    );
  };
 
 ```

Reply:
done 👍